mlir/lib/Dialect/Linalg/Transforms/Transforms.cpp

```diff
@@ -1030,6 +1042,10 @@ static Value getPackOpSourceOrPaddedSource(OpBuilder &builder,
     return input;
   }
 
+  // TODO: Support Memref PackOp. Temporarily return failure.
+  if (!packOp.hasPureTensorSemantics())
+    return packOp.getSource();
```

Comment:
The code does not agree with the comment. How about returning `Value()` when the op does not have pure tensor semantics? Please also update the function comment to reflect the change.